.github/workflows/setup-environment.yml

```diff
@@ -24,44 +28,32 @@ jobs:
           path: |
             .github/code/hypergraph
             .github/code/shared_jars
-          key: hypergraph-artifacts-${{ hashFiles('**/build.sbt', '**/src/main/scala/io/constellationnetwork/**') }}
-
-      - name: Get Version
-        id: calculate_version
-        run: |
-          BASE_VERSION="1.0.0"
-          FILES_HASH=${{ hashFiles('**/build.sbt', '**/src/main/scala/io/constellationnetwork/**') }}
-          SHORT_HASH=${FILES_HASH:0:8}
-          
-          TESSELLATION_VERSION="$BASE_VERSION-$SHORT_HASH"
-          
-          echo "TESSELLATION_VERSION=$TESSELLATION_VERSION" >> $GITHUB_ENV
-          echo "tessellation_version=$TESSELLATION_VERSION" >> $GITHUB_OUTPUT
+          key: hypergraph-artifacts-${{ hashFiles('**/build.sbt', '**/src/main/scala/io/constellationnetwork/**') }}-${{ inputs.tessellation_version }}
 
       - name: Setup Hypergraph
         if: steps.cache-hypergraph-artifacts.outputs.cache-hit != 'true'
         with:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          TESSELLATION_VERSION: ${{ steps.calculate_version.outputs.tessellation_version }}
+          TESSELLATION_VERSION: ${{ inputs.tessellation_version }}
         uses: "./.github/templates/actions/hypergraph/setup"
 
       - name: Cache SBT and Coursier
         uses: actions/cache@v4
         with:
           path: ~/.m2
-          key: sbt-${{ runner.os }}-${{ hashFiles('**/build.sbt', '**/project/*.sbt', '**/project/build.properties') }}
+          key: sbt-${{ runner.os }}-${{ hashFiles('**/build.sbt', '**/project/*.sbt', '**/project/build.properties') }}-${{ inputs.tessellation_version }}
 
       - name: Upload hypergraph artifacts to run CI tests
         uses: actions/upload-artifact@v4
         with:
-          name: hypergraph-${{ runner.os }}-${{ hashFiles('**/build.sbt', '**/project/*.sbt', '**/project/build.properties') }}
+          name: hypergraph-${{ runner.os }}-${{ hashFiles('**/build.sbt', '**/project/*.sbt', '**/project/build.properties') }}-${{ inputs.tessellation_version }}
```

Comment:
if we would extend the tessellation version by `project/*.sbt,...` we can use just the tessellation version here, for hypergraph.